anyway/parsers/infographics_data_cache_updater.py

```diff
@@ -0,0 +1,111 @@
+            f"returned value {type(db_item)}"
+            f":cause:{e.__cause__}, class:{e.__class__}"
+        )
+        return {}
+
+
+def copy_temp_into_cache():
+    num_items_cache = db.session.execute("SELECT * from infographics_data_cache").rowcount
+    num_items_temp = db.session.execute("SELECT * from infographics_data_cache_temp").rowcount
+    logging.debug(f"num items in cache: {num_items_cache}, temp:{num_items_temp}")
+    db.session.commit()
+    start = datetime.now()
+    with db.get_engine().begin() as conn:
+        conn.execute("lock table infographics_data_cache in exclusive mode")
+        logging.debug(f"in transaction, after lock")
+        conn.execute("delete from infographics_data_cache")
+        logging.debug(f"in transaction, after delete")
+        conn.execute(
+            "insert into infographics_data_cache SELECT * from infographics_data_cache_temp"
+        )
+        logging.debug(f"in transaction, after insert into")
+    logging.info(f"cache unavailable time: {str(datetime.now() - start)}")
+    num_items_cache = db.session.execute("SELECT * from infographics_data_cache").rowcount
+    num_items_temp = db.session.execute("SELECT * from infographics_data_cache_temp").rowcount
+    logging.debug(f"num items in cache: {num_items_cache}, temp:{num_items_temp}")
+    db.session.execute("truncate table infographics_data_cache_temp")
+    db.session.commit()
+    num_items_cache = db.session.execute("SELECT * from infographics_data_cache").rowcount
+    num_items_temp = db.session.execute("SELECT * from infographics_data_cache_temp").rowcount
+    logging.debug(f"num items in cache: {num_items_cache}, temp:{num_items_temp}")
+    db.session.commit()
+
+
+def build_cache_into_temp():
+    start = datetime.now()
+    db.session.query(InfographicsDataCacheTemp).delete()
+    db.session.commit()
+    db.get_engine().execute(
+        InfographicsDataCacheTemp.__table__.insert(),  # pylint: disable=no-member
+        [
+            {
+                "news_flash_id": new_flash.get_id(),
+                "years_ago": y,
+                "data": anyway.infographics_utils.create_infographics_data(new_flash.get_id(), y),
+            }
+            for y in CONST.INFOGRAPHICS_CACHE_YEARS_AGO
+            for new_flash in db.session.query(NewsFlash).filter(NewsFlash.accident).all()
+        ],
+    )
+    logging.info(f"cache rebuild took:{str(datetime.now() - start)}")
+
+
+def get_cache_info():
+    cache_items5 = (
+        db.session.query(InfographicsDataCache).filter(InfographicsDataCache.years_ago == 5).count()
+    )
+    cache_items3 = (
+        db.session.query(InfographicsDataCache).filter(InfographicsDataCache.years_ago == 3).count()
+    )
+    tmp_items5 = (
+        db.session.query(InfographicsDataCacheTemp)
+        .filter(InfographicsDataCacheTemp.years_ago == 5)
+        .count()
+    )
+    tmp_items3 = (
```

Comment:
@ziv17 Do you think we need also items for 1 and 8 years ago?

Reply:
Actually this is more of a debug print I added when I suspected I have different number of entries for 3 and 5. I think I will remove it.